src/Exception/MailCancelledException.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AcMailer\Exception;
+
+use RuntimeException;
+
+class MailCancelledException extends RuntimeException implements ExceptionInterface
+{
```

Comment:
I usually prefer to expose named constructors from exceptions, so that the instantiation with arbitrary messages is not spread around.

So, I'd suggest having something like this:

```php
public static function fromEvent(string $eventName): self
{
    return self(sprintf('Email cancelled from "%s" event listener', $eventName));
}
```